### DISCUSSION TITLE
Having trouble restoring an OKD-affected device using multiple methods

### DISCUSSION THREAD

**Discussion by eblanshey** [dangowrt/owrt-ubi-installer#192] [Q&A] [answered]
Hi, I have 3 Belkin RT3200 devices, two of which are running a version affected by OKD. They were inconsistently booting. 

**Prelude:** To start, I followed the 4 bullet points [instructions in the wiki](https://openwrt.org/toh/linksys/e8450#recovery_from_live_openwrt_mainmaster_snapshot_with_fip_in_ubi_layout) at the time of this writing to fix the issue. The gist is: rewrite FIP; sysupgrade to the latest image from the [snapshot repository](https://downloads.openwrt.org/snapshots/targets/mediatek/mt7622/); install kmod-mtd-rw; use mtd to rewrite BL2.

Next I was having some troubles with the new snapshot, so I tried going two different routes, each having their own issue.

**Attempt 1, using latest image:** I flashed the unsigned installer from the v1.1.3 release "The defeat of OKD". The LuCi interface didn't come up, so I booted with reset, then flashed the latest "ubi-squashfs-sysupgrade" snapshot image from the official snapshot repository. Again, LuCi does not come up. When I SSH into it, httpd isn't running. The router is successfully providing internet via ethernet, though.

**Attempt 2, using stable release:** Since LuCi wasn't working, I booted holding reset button, and flashed the installer from the last stable release this time. Then immediately went in and flashed the sysupgrade image from the same stable release's assets. This time LuCi comes up, but... the 5ghz radio is missing, and I couldn't figure out how restore it. I made sure it booted normally (I read that 5ghz can be missing if it's in restore mode.) My 1st router which is working for me is somehow on version 22.03.5, whereas the stable release installs 22.03.3 -- I wonder if the version difference is causing the radio to be missing.

I'm at a loss of what to try next and would appreciate suggestions. Thank you.

**Comment by fda77** [comment]
The 1.1.3 installer is buggy and briked my device too. With serial connection this helped me: https://forum.openwrt.org/t/belkin-rt3200-linksys-e8450-wifi-ax-discussion/94302/5927

**Reply by eblanshey** [reply]
1.1.3 installer didn't brick my device, but those steps look very similar to what I already did, but perhaps using different version numbers. I guess I'll give it another shot tonight. (You say 1.1.3 doesn't work but those steps do use the 1.1.3 installer at the end...)

**Comment by dangowrt** [comment, marked as answer]
1. snapshot images comes without LuCI by definition. If you want LuCI you need to install it, ie. login via SSH and
```
opkg update
opkg install luci-ssl
```

2. Once you run the v1.1.3 installer there is no easy way back. You cannot just run the v1.0.x to revert back to OpenWrt 23.05. OpenWrt 23.05 won't work on systems which previously run the v1.1.3 installer. I recommend you to just you snapshot build and install LuCI manually until there is going to be the next official release which is going to be based on v1.1.x installer as well.

**Reply by eblanshey** [reply]
I wasn't aware the snapshots didn't come with LuCi, thank you! Indeed that solved that issue. Unfortunately I've come to the next problem while setting it up: the radio types aren't recognized, nor does scanning pick up any signals. See the image below. It's supposed to detect that one of them is 2.4ghz (bgn) and the other 5ghz (acaxn). Any ideas why?

![image](https://github.com/user-attachments/assets/8f1665a4-b282-4a45-8528-bc5f014d7ad8)

I used the latest `openwrt-mediatek-mt7622-linksys_e8450-ubi-squashfs-sysupgrade.itb` image.

**Reply by eblanshey** [reply]
All I had to do was to enable the wireless networks to make the radios start working. All is well now, thank you Dan!
